### PR TITLE
Can emag artifact crusher

### PR DESCRIPTION
**Pull request by SpeltIncorrectyl**
## About the PR
Can now emag artifact crusher.
Normally when someone is inside an artifact crusher, they will still be able to leave when it is crushing. They can just open the door and the machine will halt.
When emagged, the artifact crusher will lock shut for the duration of the crushing. It will not trap someone inside forever - once the crushing is over the locks will release. However the person inside is probably dead by the time this happens.

## Why / Balance
Someone who is stuck inside when the machine starts crushing cannot escape by themselves, and will be gibbed.
However the gibbing takes 10 seconds, and someone outside can unbolt the machine, which cause the crushing to stop, and the door will unlock and they can escape.
The artifact crusher is a large conspicuous machine, and examining it will tell you if it is emagged (it will say "Autolocks enabled", see the video).
It is far less dangerous than an emagged recycler, which will instagib anyone who walks into it,

## Technical details
* Added AutoLock field to ArtifactCrusherComponent.
* SharedArtifactCrusherSystem susbcribes to OnStorageOpenAttempt and cancels it is the machine is crushing and the autolocks are enabled.
* SharedArtifactCrusherSystem subscribes to OnExamineEvent to add examine text about the crusher's autolocks being enabled or disabled.
* SharedArtifactCrusherSystem subscribes to GotEmaggedEvent to set AutoLock to true when emagged.

## Media

https://github.com/space-wizards/space-station-14/assets/66873282/756dacd0-1898-4ad3-b746-a000660da4a1
- [X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
* ArtifactCrusherComponent has new field, AutoLock. This determines if the machine can be opened while it is crushing.

**Changelog**
:cl:
- add: Emagging the artifact crusher now stops it from being opened while it is crushing.

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Based murder pill

**Review by Brandon-Huu — APPROVED** [review]
This is amazing. Thank you.

**Comment by MrToofters** [comment]
Absolutely cool way to murder someone

File: Content.Shared/Xenoarchaeology/Equipment/SharedArtifactCrusherSystem.cs [hunk 1]
```diff
@@ -33,6 +38,23 @@ private void OnStorageAfterOpen(Entity<ArtifactCrusherComponent> ent, ref Storag
...
+
+    private void OnStorageOpenAttempt(Entity<ArtifactCrusherComponent> ent, ref StorageOpenAttemptEvent args)
+    {
+        if (ent.Comp.AutoLock && ent.Comp.Crushing)
+            args.Cancelled = true;
+    }
+
+    private void OnExamine(Entity<ArtifactCrusherComponent> ent, ref ExaminedEvent args)
+    {
+        args.PushMarkup(ent.Comp.AutoLock ? Loc.GetString("artifact-crusher-examine-autolocks") : Loc.GetString("artifact-crusher-examine-no-autolocks"));
```
Comment (EmoGarbage404) [inline Content.Shared/Xenoarchaeology/Equipment/SharedArtifactCrusherSystem.cs:55]: these locale strings aren't in the commits. Did you forget to add/commit them before you made the PR?

File: Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactCrusherSystem.cs [hunk 2]
```diff
@@ -61,11 +64,14 @@ private void OnPowerChanged(Entity<ArtifactCrusherComponent> ent, ref PowerChang
...
     {
-        var (_, crusher, _) = ent;
+        var (uid, crusher, _) = ent;
 
         if (crusher.Crushing)
             return;
 
+        if (crusher.AutoLock)
+            _popup.PopupEntity(Loc.GetString("artifact-crusher-autolocks-enable"), uid);
+
```
Comment (EmoGarbage404) [inline Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactCrusherSystem.cs:74]: same deal here. As a note, having a popup play every time seems really annoying so maybe just have it on the attempt open event. That way the person inside can see that it's locked for all glorious 10 seconds they're getting crushed.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Shared/Xenoarchaeology/Equipment/SharedArtifactCrusherSystem.cs (hunk 1 above)
Comment (SpeltIncorrectyl) [reply Content.Shared/Xenoarchaeology/Equipment/SharedArtifactCrusherSystem.cs:55]: Forgot to add the file to git, committed it now.

**Review by EmoGarbage404 — APPROVED** [review]

**Pull request merged**